src/main/thrift/parquet.thrift

```diff
@@ -885,6 +973,44 @@ struct ColumnChunk {
   9: optional binary encrypted_column_metadata
 }
 
+struct ColumnChunkV3 {
+  /** File where column data is stored. **/
+  1: optional string file_path
+
+  /** Byte offset in file_path to the ColumnChunkMetaDataV3, optionally encrypted
+   ** CHANGED from v1: renamed to metadata_file_offset
+   **/
+  2: required i64 metadata_file_offset
+
+  /** NEW from v1: Byte length in file_path of ColumnChunkMetaDataV3, optionally encrypted
+   **/
+  3: required i32 metadata_file_length
```

Comment:
Thank you :) 

The naming is a little awkward...I read these as "metadata-file offset"/"metadata-file length". Perhaps instead just "metadata_offset" and "metadata_length"?

Aside: just curious, but in the current format how do you make use of the `file_offset`? Is there a way to deduce the length of the metadata? Or do you have to use a file based reader and seek to the offset?

Reply:
BTW, do we still want to put a copy `ColumnMetaData` at the end of column chunk and another copy here at `4: optional binary encoded_metadata`? I know it is good to keep backward compatibility. Does any implementation actually read it from the end of column chunk?